DirectProgramming/Fortran/DenseLinearAlgebra/optimize-integral/README.md

```diff
@@ -0,0 +1,194 @@
+# `Optimization Integral`
+ 
+This sample is designed to illustrate compiler optimization features and programming concepts.
+
+This program computes the integral (area under the curve) of a user-supplied function 
+over an interval in a stepwise fashion. 
+The interval is split into segments, and at each segment position the area of a rectangle 
+is computed whose height is the value of sine at that point and the width is the segment width. 
+The areas of the rectangles are then summed.
+
+The process is repeated with smaller and smaller width rectangles, 
+more closely approximating the true value.
+
+The source for this program also demonstrates recommended Fortran coding practices.
+
+| Optimized for                     | Description
+|:---                               |:---
+| OS                                | macOS* with Xcode* installed 
+| Software                          | Intel&reg; oneAPI Intel® Fortran Compiler (Beta)
+| What you will learn               | Optimization using the Intel® Fortran compiler
+| Time to complete                  | 15 minutes
+
+## Purpose
+
+The Intel® Fortran Compiler can optimize applications for performance.  The primary compiler option is -O followed by a numeric optimizaiton "level" from 0 requesting no optimization to 3, which requests all compiler optimizations for the application. The -O optimizaition levels are:
+
+   * O0 - No optimizations
+   * O1 - Enables optimizations for speed and disables some optimizations that increase code size and affect speed.
+   * O2 - Enables optimizations for speed. This is the generally recommended optimization level. Vectorization is enabled at O2 and higher levels.
+   * O3 - Performs O2 optimizations and enables more aggressive loop transformations such as Fusion, Block-Unroll-and-Jam, and collapsing IF statements.
+
+Read the [Intel® Fortran Compiler Developer Guide and Reference][1]
+[1]: https://software.intel.com/content/www/us/en/develop/documentation/fortran-compiler-developer-guide-and-reference/top.html "Intel® Fortran Compiler Developer Guide and Reference" 
+for more information about these options.
+
+Some of these compiler optimizations use features and options that can 
+restrict program execution to specific architectures.  
+
+
+## License  
+This code sample is licensed under MIT license  
+
+## Building the `Fortran Optimization` sample
+  
+Use the one of the following compiler options:
+
+
+### macOS* : -O0 -O1, -O2, -O3 
+
+### STEP 1: Build and run with -O0
+cd optimize_samples 
```

Comment:
Change the following
cd optimize_samples
to
cd optimize-integral

Reply:
done, will push 8/24